### ISSUE TITLE
Migration to Qt6

### ISSUE THREAD

**Issue by jrozner** [ashirt-ops/ashirt#72]
Qt has released the new major version 6. It sounds like there's still a handful of features that are not yet available in 6 yet but none of them seem relevant to us. We should evaluate what it will take to support Qt6 and get it building. Given that most Linux distros are unlikely to be shipping it yet, we don't want to move away from Qt5 yet but we do want to make sure that we're developing with the future in mind.

Some big changes include:
* shift in upstream build system from qmake to cmake (we probably want to move to cmake as well)
* upstream has moved to c++17 (we should as well)

**Comment by JoelAtDeluxe** [comment]
An initial PR that adds compilation support is ready, though a true switch has not yet been done.

**Comment by jrozner** [comment]
The official building for MacOS depends on homebrew currently. We're likely not going to be able to actually migrate to Qt6 until [this](https://github.com/Homebrew/homebrew-core/pull/68134) merges. That's fine though, we're not in a rush to get onto Qt6. Just want to make sure that when we're ready it's an easy transition.

**Comment by jrozner** [comment]
Looks like the previous PR has been closed and replaced with https://github.com/Homebrew/homebrew-core/pull/70826

**Comment by jrozner** [comment]
Qt 6 is generally available for macos and most distros now it looks like. We're not using any of the extensions that aren't ready yet so we might as well move over. Let's do the cmake transition in another PR but move to cmake + C++17 at the same time.

**Comment by jrozner** [comment]
We have working Qt6 builds of all platforms. Polish work is probably needed across the board before we can release 1.2.0 but this work is effectively done. Need to start running through manual testing and identifying blockers.
